rasa/telemetry.py

```diff
@@ -0,0 +1,553 @@
+import asyncio
+import async_generator
+from datetime import datetime
+import functools
+from functools import wraps
+import hashlib
+import json
+import logging
+import multiprocessing
+import os
+from pathlib import Path
+import platform
+from subprocess import CalledProcessError, STDOUT, check_output  # skipcq:BAN-B404
+import sys
+import textwrap
+from typing import Any, Callable, Dict, Optional, Text
+import uuid
+
+import aiohttp
+from terminaltables import SingleTable
+
+import rasa
+from rasa.shared.constants import DOCS_URL_TELEMETRY
+from rasa.shared.importers.importer import TrainingDataImporter
+import rasa.shared.utils.io
+from rasa.constants import (
+    CONFIG_FILE_TELEMETRY_KEY,
+    CONFIG_TELEMETRY_DATE,
+    CONFIG_TELEMETRY_ENABLED,
+    CONFIG_TELEMETRY_ID,
+)
+from rasa.utils import common as rasa_utils
+import rasa.utils.io
+
+logger = logging.getLogger(__name__)
+
+SEGMENT_ENDPOINT = "https://api.segment.io/v1/track"
+
+TELEMETRY_ENABLED_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_ENABLED"
+TELEMETRY_DEBUG_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_DEBUG"
+
+# the environment variable can be used for local development to set a test key
+# e.g. `RASA_TELEMETRY_WRITE_KEY=12354 rasa train`
+TELEMETRY_WRITE_KEY_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_WRITE_KEY"
+
+TELEMETRY_ID = "metrics_id"
+TELEMETRY_ENABLED_BY_DEFAULT = True
+
+# if one of these environment variables is set, we assume to be running in CI env
+CI_ENVIRONMENT_TELL = [
+    "bamboo.buildKey",
+    "BUILD_ID",
+    "BUILD_NUMBER",
+    "BUILDKITE",
+    "CI",
+    "CIRCLECI",
+    "CONTINUOUS_INTEGRATION",
+    "GITHUB_ACTIONS",
+    "HUDSON_URL",
+    "JENKINS_URL",
+    "TEAMCITY_VERSION",
+    "TRAVIS",
+]
+
+# If updating or creating a new event, remember to update
+# https://rasa.com/docs/rasa/telemetry
+TRAINING_STARTED_EVENT = "Training Started"
+TRAINING_COMPLETED_EVENT = "Training Completed"
+TELEMETRY_DISABLED = "Telemetry Disabled"
+
+
+def print_telemetry_reporting_info() -> None:
+    """Print telemetry information to std out."""
+    message = textwrap.dedent(
```

Comment:
do we really need `textwrap`? You should be able to remove the indent to the left of the message without `black` complaining

Reply:
in my opinion it makes it harder to read if a block of text is not properly indented like the rest of the code